app/src/main/java/com/ddoczi/tasky/agenda/presentation/detail/task/TaskDetailViewModel.kt

```diff
@@ -0,0 +1,17 @@
+package com.ddoczi.tasky.agenda.presentation.detail.task
+
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import javax.inject.Inject
+
+@HiltViewModel
+class TaskDetailViewModel @Inject constructor(
+
+): ViewModel() {
+    private val _state = MutableStateFlow(TaskDetailState())
+    val state = _state.asStateFlow()
+
+    fun onEvent(event: TaskDetailEvent) {}
+}
```

Comment:
Especially in the ViewModels, you'll have a lot of duplication when having one VM per agenda item type, since all the state updating logic needs to be duplicated